portal-kernel/src/com/liferay/portal/kernel/util/comparator/RoleTitleComparator.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.kernel.util.comparator;
+
+import com.liferay.portal.kernel.model.Role;
+import com.liferay.portal.kernel.util.OrderByComparator;
+
+/**
+ * @author Balazs Breier
+ */
+public class RoleTitleComparator extends OrderByComparator<Role> {
+
+	public static final String ORDER_BY_ASC = "Role_.title ASC";
+
+	public static final String ORDER_BY_DESC = "Role_.title DESC";
+
+	public static final String[] ORDER_BY_FIELDS = {"title"};
+
+	public RoleTitleComparator() {
+		this(false);
+	}
+
+	public RoleTitleComparator(boolean ascending) {
+		_ascending = ascending;
+	}
+
+	@Override
+	public int compare(Role role1, Role role2) {
+		String title1 = role1.getTitle();
+		String title2 = role2.getTitle();
+		String key1 = role1.getName();
+		String key2 = role2.getName();
+
+		int value;
+
+		if (role1.getTitle().isEmpty()) {
```

Comment:
Please use `Validator.isNull(role1.getTitle())` and also don't chain methods (so in this case you don't have to use `role1.getTitle().isEmpty` but `title1.isEmpty()`